index.js

```diff
@@ -1,5 +1,6 @@
 'use strict';
-const re = '[\\w.+-]+\\@[\\w-]+\\.[\\w]+';
+
+const re = '[\\w][\\w.+-]*[\\w\\d]\\@[\\w-]+(\\.[\\w]+)+';
```

Comment:
1) \d after \w is redundant as \w equals numbers as well.
2) We don't want to capture just the extension, make it a non-capturing group.